File: Solutions/0301-400/387. First Unique Character in a String/solution2.java
class Solution {
    public int firstUniqChar(String s) {
        int n=s.length();
        int[] freq=new int[26];
        for(int i=0;i<n;i++){
            char ch=s.charAt(i);
            freq[ch-'a']++;
        }
        for(int i=0;i<n;i++){
            char ch=s.charAt(i);
            if(freq[ch-'a']==1){
                return i;
            }
        }
        return -1;
    }
}

File: Solutions/1600-1699/1679. Max Number of K-Sum Pairs/Solution2.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        freq = Counter()
        res = 0

        for i in nums:
            compliment = k - i
            if freq[compliment] > 0:
                res += 1
                freq[compliment] -= 1
            else:
                freq[i] += 1
        
        return res
